### PR TITLE
puan-rspy 0.2.8

### PR DESCRIPTION
**Pull request by znittzel**
Fixed plog.errors function
maz-0.0.2 implemented
solve methods returns dict instead of SolutionVariable SolutionVariable is dead

### CODE REVIEW

File: puan/logic/plog/__init__.py [hunk 1]
```diff
@@ -339,6 +340,10 @@ def errors(self) -> typing.List[PropositionValidationError]:
 
         """
             Checks this proposition and returns a list of :class:`PropositionValidationError`'s. 
+            If there's at least one error in result, it is because one or more of following:
+                - It exist at least one circular variable dependency.
+                - A variable depends directly on at least two other variables with identical id's.
```
Comment (ourmoa) [inline puan/logic/plog/__init__.py:345]:
```suggestion
                - A variable depends directly on two or more variables with identical ids.
```

File: puan/logic/plog/__init__.py [hunk 2]
```diff
@@ -339,6 +340,10 @@ def errors(self) -> typing.List[PropositionValidationError]:
...
+                - There exists at least two variables sharing id but has different bounds.
```
Comment (ourmoa) [inline puan/logic/plog/__init__.py:346]:
```suggestion
                - There exists two or more variables with identical ids but with different bounds.
```

File: puan/logic/plog/__init__.py [hunk 3]
```diff
@@ -355,41 +360,138 @@ def errors(self) -> typing.List[PropositionValidationError]:
...
+                    # variable dependecy somewhere in the model
+                    maz.fnexcept(
+                        maz.compose(
+                            operator.not_,
+                            functools.partial(
+                                operator.eq,
+                                None,
+                            ),
+                            operator.methodcaller("prepare"),
+                            graphlib.TopologicalSorter,
```
Comment (ourmoa) [inline puan/logic/plog/__init__.py:386]: How is this different from the old toposort? (Just curious)

File: tests/test_puan.py [hunk 4]
```diff
@@ -2667,22 +2658,42 @@ def test_proposition_errors_function():
...
+    # Sharing same id on same level and same sub propositions
     actual = pg.All(
         pg.AtMost(0, [puan.variable('a')], variable=''),
+        pg.AtMost(0, [puan.variable('a')], variable='')
+    ).errors()
+    assert len(actual) == 1
+    assert actual[0] == pg.PropositionValidationError.NON_UNIQUE_SUB_PROPOSITION_SET
+
+    # Should return error since we have two B's with different
+    # childs
```
Comment (ourmoa) [inline tests/test_puan.py:2690]:
```suggestion
    # children
```

**Review by ourmoa — CHANGES_REQUESTED** [review]

File: puan/logic/plog/__init__.py (hunk 3 above)
Comment (znittzel) [reply puan/logic/plog/__init__.py:386]: This one is actually python's package and it also fails for these cases A -> A (which it should and the old one didn't)

**Review by ourmoa — APPROVED** [review]

**Pull request merged**